exchanges/btcmarkets/btcmarkets.go

```diff
@@ -53,316 +84,579 @@ type BTCMarkets struct {
+}
 
-	err := b.SendAuthenticatedRequest(http.MethodGet, btcMarketsAccountBalance, nil, &balance)
-	if err != nil {
-		return nil, err
+// FetchDepositAddress gets deposit address for the given asset
+func (b *BTCMarkets) FetchDepositAddress(assetName string, before, after, limit int64) (DepositAddress, error) {
+	var resp DepositAddress
+	if (before > 0) && (after >= 0) {
+		return resp, errors.New("BTCMarkets only supports either before or after, not both")
 	}
-
-	// All values are returned in Satoshis, even for fiat currencies.
-	for i := range balance {
-		balance[i].Balance /= common.SatoshisPerBTC
-		balance[i].PendingFunds /= common.SatoshisPerBTC
+	params := url.Values{}
+	params.Set("assetName", assetName)
+	if before > 0 {
+		params.Set("before", strconv.FormatInt(before, 10))
+	}
+	if after >= 0 {
+		params.Set("after", strconv.FormatInt(after, 10))
+	}
+	if limit > 0 {
+		params.Set("limit", strconv.FormatInt(limit, 10))
 	}
-	return balance, nil
+	return resp, b.SendAuthenticatedRequest(http.MethodGet,
+		common.EncodeURLValues(btcMarketsAddresses, params),
+		nil,
+		&resp)
 }
 
-// GetTradingFee returns the account's trading fee for a currency pair
-func (b *BTCMarkets) GetTradingFee(base, quote currency.Code) (TradingFee, error) {
-	var tradingFee TradingFee
-	path := fmt.Sprintf(btcMarketsTradingFee, base, quote)
-	return tradingFee, b.SendAuthenticatedRequest(http.MethodGet, path, nil, &tradingFee)
+// GetWithdrawalFees gets withdrawal fees for all assets
+func (b *BTCMarkets) GetWithdrawalFees() ([]WithdrawalFeeData, error) {
+	var resp []WithdrawalFeeData
+	return resp, b.SendHTTPRequest(btcMarketsAPIURL+btcMarketsAPIVersion+btcMarketsWithdrawalFees,
+		&resp)
 }
 
-// WithdrawCrypto withdraws cryptocurrency into a designated address
-func (b *BTCMarkets) WithdrawCrypto(amount float64, currency, address string) (string, error) {
-	newAmount := int64(amount * float64(common.SatoshisPerBTC))
+// ListAssets lists all available assets
+func (b *BTCMarkets) ListAssets() ([]AssetData, error) {
+	var resp []AssetData
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, btcMarketsAssets, nil, &resp)
+}
 
-	req := WithdrawRequestCrypto{
-		Amount:   newAmount,
-		Currency: strings.ToUpper(currency),
-		Address:  address,
+// GetTransactions gets trading fees
+func (b *BTCMarkets) GetTransactions(assetName string, before, after, limit int64) ([]TransactionData, error) {
+	if (before > 0) && (after >= 0) {
+		return nil, errors.New("BTCMarkets only supports either before or after, not both")
 	}
-
-	resp := Response{}
-	err := b.SendAuthenticatedRequest(http.MethodPost,
-		btcMarketsWithdrawCrypto,
-		req,
-		&resp)
-	if err != nil {
-		return "", err
+	var resp []TransactionData
+	params := url.Values{}
+	if assetName != "" {
+		params.Set("assetName", assetName)
 	}
-
-	if !resp.Success {
-		return "", errors.New(resp.ErrorMessage)
+	if before > 0 {
+		params.Set("before", strconv.FormatInt(before, 10))
 	}
+	if after >= 0 {
```

Comment:
Any reason why this is `>=` instead of just `>`? It means it will always be included in the request

Reply:
I've seen in the wrapper you call everything with `-1`, so I'm okay with it